nmrfx-processor-gui/src/main/java/org/nmrfx/processor/gui/MainApp.java

```diff
@@ -92,7 +92,7 @@ public static void closeAll() {
             chart.clearDataAndPeaks();
             chart.clearAnnotations();
         }
-        List<FXMLController> controllers = FXMLController.getControllers();
+        List<FXMLController> controllers = new ArrayList<>(FXMLController.getControllers());
         // Don't close the first controller that matches with the main stage, Note this first controller is not
         // necessarily the active controller
         for (int index = 1; index < controllers.size(); index++) {
```

Comment:
controllers.size was decreasing each loop due to the close call in each loop, so not all the controllers were closed,